ocaml/xapi/daemon_manager.ml

```diff
@@ -63,14 +63,29 @@ module Make(D : DAEMON) = struct
 		end
 	| Function f -> f ()
 
-	let with_daemon_stopped f =
+	let start = D.start
+
+	let stop ?timeout () =
+		match timeout with
+		| Some t -> begin
```

Comment:
Moving the `begin .. end` so it's just around the exception handler means the compiler thinks `| None -> ` is an exception match instead of a top level match, and it doesn't compile (since `None` isn't an exception). Is that what you meant?